toolkit/pythonit_toolkit/api/service_client.py

```diff
@@ -0,0 +1,47 @@
+from dataclasses import dataclass
+from typing import Any, Dict, Optional
+
+import httpx
+from pythonit_toolkit.headers import SERVICE_JWT_HEADER
+from pythonit_toolkit.pastaporto.tokens import generate_token
+
+
+@dataclass
+class ServiceResponse:
+    errors: Optional[Dict[str, Any]]
+    data: Optional[Dict[str, Any]]
+
+
+class ServiceClient:
+    def __init__(
+        self,
+        url: str,
+        issuer: str,
+        audience: str,
```

Comment:
we can keep it as it is for now, but I think we should make `url` and `audience` simpler, we are contacting a service so this __init__ could technically just take a `service_name` as property, and find the URL in some other way? what do you think? I just need to think of a way to have the list of URLs, we could take them from env variables but I am not a fan of forcing env variables at the toolkit level

Reply:
Agree

Reply:
```suggestion
        service_name: str,
        url: str,
        caller: str,
```

service_name is the destination (audience in the JWT)
url is the URL of the destination's service
caller is the service calling the destination (issuer in the JWT)

I would use those names in the service client, since the names of the JWT properties do not make sense in the client context